src/detector.ts

```diff
@@ -132,12 +131,19 @@ async function processor(data: ProductEvent[]): Promise<ProcessorResult> {
     retry: new Set(),
   };
   const promises = [];
+  const config: Config = {
+    apiKey: window.TS.token,
+    host: window.TS.url,
+  };
   for (const entry of data) {
     promises.push(
-      reportEvent(getApiPayload(entry), window.TS)
+      reportEvent(config, getApiPayload(entry))
         .then((result) => {
-          const q = result.retry ? r.retry : r.done;
-          q.add(entry.id);
+          if (result.ok) {
+            r.done.add(entry.id);
+          } else {
+            r.retry.add(entry.id);
```

Comment:
Note that not all response code should be retried. Only 429 and 5xx.

Reply:
Will adapt to have it correctly coming from the sdk! Thank you very much